docs/subnets/introduction-to-vm.md

```diff
@@ -0,0 +1,363 @@
+
+Every VM should implement the following interfaces:
+
+### `block.ChainVM`
+
+To reach a consensus on linear blockchains (as opposed to DAG blockchains), Avalanche uses the Snowman consensus engine. To be compatible with Snowman, a VM must implement the `block.ChainVM` interface, which can be accessed from [AvalancheGo repository](https://github.com/ava-labs/avalanchego/blob/v1.7.4/snow/engine/snowman/block/vm.go).
+
+```go title="/snow/engine/snowman/block/vm.go"
+// ChainVM defines the required functionality of a Snowman VM.
+//
+// A Snowman VM is responsible for defining the representation of the state,
+// the representation of operations in that state, the application of operations
+// on that state, and the creation of the operations. Consensus will decide on
+// if the operation is executed and the order operations are executed.
+//
+// For example, suppose we have a VM that tracks an increasing number that
+// is agreed upon by the network.
+// The state is a single number.
+// The operation is setting the number to a new, larger value.
+// Applying the operation will save to the database the new value.
+// The VM can attempt to issue a new number, of larger value, at any time.
+// Consensus will ensure the network agrees on the number at every block height.
+type ChainVM interface {
+	common.VM
+	Getter
+	Parser
+
+	// Attempt to create a new block from data contained in the VM.
+	//
+	// If the VM doesn't want to issue a new block, an error should be
+	// returned.
+	BuildBlock() (snowman.Block, error)
+
+	// Notify the VM of the currently preferred block.
+	//
+	// This should always be a block that has no children known to consensus.
+	SetPreference(ids.ID) error
+
+	// LastAccepted returns the ID of the last accepted block.
+	//
+	// If no blocks have been accepted by consensus yet, it is assumed there is
+	// a definitionally accepted block, the Genesis block, that will be
+	// returned.
+	LastAccepted() (ids.ID, error)
+}
+
+// Getter defines the functionality for fetching a block by its ID.
+type Getter interface {
+	// Attempt to load a block.
+	//
+	// If the block does not exist, an error should be returned.
+	//
+	GetBlock(ids.ID) (snowman.Block, error)
+}
+
+// Parser defines the functionality for fetching a block by its bytes.
+type Parser interface {
+	// Attempt to create a block from a stream of bytes.
+	//
+	// The block should be represented by the full byte array, without extra
+	// bytes.
+	ParseBlock([]byte) (snowman.Block, error)
+}
+```
+
+### `common.VM`
+
+`common.VM` is a type that every `VM`, whether a DAG or linear chain, must implement.
+
+You can see the full file from [here.](https://github.com/ava-labs/avalanchego/blob/v1.7.4/snow/engine/common/vm.go)
```

Comment:
should this link be to the master branch or v1.7.4? If master branch, please search and replace all v1.7.4.